eth/tracers/internal/tracers/call_tracer_parity.js

```diff
@@ -50,11 +50,30 @@
 		return true;
 	},
 
+	isStepSuccessor: false,
+
+	// isStepNeeded (optional) is used for optimisation reasons, checking if JS context
+	// needs to be loaded and step method to be called for current opcode.
+	isStepNeeded: function(log) {
+		// We only care about system opcodes
+		if ((log.op.toNumber() & 0xf0) == 0xf0) {
+			this.isStepSuccessor = true;
+			return true;
+		} else if (this.isStepSuccessor) {
+			this.isStepSuccessor = false;
+			return true;
+		} else if (log.getDepth() == this.callstack.length - 1) {
+			this.isStepSuccessor = false;
+			return true;
+		}
+		return false;
+	},
+
 	// step is invoked for every opcode that the VM executes.
 	step: function(log, db) {
 		// Capture any errors immediately
 		var error = log.getError();
-		if (typeof error !== "undefined") {
+		if (error !== undefined) {
```

Comment:
There's a lot of these `undefined` vs. `"undefined"` comparison changes, and they're also made at https://github.com/etclabscore/core-geth/pull/256/files. Is this important to both proposals?  Or could these be extracted to a separate refactoring/optimization change set?

Reply:
They are on both proposals.It could be on a separate PR, but I didn't feel the need to separate this single commit. Let me know if you prefer it that way.